plugins/systemstat.koplugin/main.lua

```diff
@@ -180,37 +184,50 @@ function SystemStat:appendProcessInfo()
     end
 
     if #t < 23 then return end
-    n1 = tonumber(t[23])
-    if n1 ~= nil then
-        self:put({_("  Virtual memory (MB)"), string.format("%.2f", n1 / 1024 / 1024)})
-    end
-
-    if #t < 24 then return end
-    n1 = tonumber(t[24])
-    if n1 ~= nil then
-        self:put({_("  RAM usage (MB)"), string.format("%.2f", n1 / 256)})
+    local virtual_mem = tonumber(t[23])
+    local ram_usage = tonumber(t[24]) -- will give nil if not avail
+
+    if virtual_mem ~= nil then
+        local key = _("  Virtual memory") .. (ram_usage and SEP .._("RAM usage") or "")
+        local value = util.getFriendlySize(virtual_mem, false) ..
+            (ram_usage and (SEP .. util.getFriendlySize(ram_usage * 4 * 1024, false)) or "")
+        self:put({key, value})
     end
 end
 
 function SystemStat:appendStorageInfo()
     if self.storage_filter == nil then return end
 
-    local std_out = io.popen(
-        "df -h | sed -r 's/ +/ /g' | grep " .. self.storage_filter ..
-        " | sed 's/ /\\t/g' | cut -f 2,4,5,6"
-    )
+    local df_commands = {
+        -- first choice: SI-prefixes
+        "df -H | sed -r 's/ +/ /g' | grep " .. self.storage_filter .. " | sed 's/ /\\t/g' | cut -f 2,4,5,6",
+        -- second choice: binary-prefixes
+        "df -h | sed -r 's/ +/ /g' | grep " .. self.storage_filter .. " | sed 's/ /\\t/g' | cut -f 2,4,5,6",
+        -- use busybox if available (on some Android devices)
+        "busybox df -h | busybox sed -r 's/ +/ /g' | busybox grep " .. self.storage_filter .. " | busybox sed 's/ /\\t/g' | busybox cut -f 2,4,5,6",
+    }
+
+    local std_out
+    local command_index = 1
+    while not std_out and command_index <= #df_commands+1 do
+        std_out = io.popen( df_commands[command_index] )
+        command_index = command_index + 1
+    end
```

Comment:
Or `ipairs(df_commands)` and `break` when you get a std_out that is not bad ? Which is our classic idiom for this kind of loop.